runtime-modules/content/src/tests/mock.rs

```diff
@@ -536,7 +536,59 @@ impl Default for ExtBuilder {
     }
 }
 
+// TODO(post mainnet?): authomatically set block number = 1
 impl ExtBuilder {
+    pub fn with_creator_royalty_bounds(self, min_creator_royalty: Perbill, max_creator_royalty: Perbill) -> Self {
+        Self {
+            min_creator_royalty,
+            max_creator_royalty,
+            ..self
+        }
+    }
+    /// test externalities + initial balances allocation
+    pub fn build_with_balances(
+        self,
+        balances: Vec<(AccountId, BalanceOf<Test>)>,
+    ) -> sp_io::TestExternalities {
+        let mut t = frame_system::GenesisConfig::default()
+            .build_storage::<Test>()
+            .unwrap();
+
+        balances::GenesisConfig::<Test> { balances }
+        .assimilate_storage(&mut t)
+        .unwrap();
+
+        // the same as t.top().extend(GenesisConfig::<Test> etc...)
+        crate::GenesisConfig::<Test> {
+            next_channel_category_id: self.next_channel_category_id,
+            next_channel_id: self.next_channel_id,
+            next_video_id: self.next_video_id,
+            next_curator_group_id: self.next_curator_group_id,
+            max_cashout_allowed: self.max_cashout_allowed,
+            min_cashout_allowed: self.min_cashout_allowed,
+            channel_cashouts_enabled: self.channel_cashouts_enabled,
+            min_auction_duration: self.min_auction_duration,
+            max_auction_duration: self.max_auction_duration,
+            min_auction_extension_period: self.min_auction_extension_period,
+            max_auction_extension_period: self.max_auction_extension_period,
+            min_bid_lock_duration: self.min_bid_lock_duration,
+            max_bid_lock_duration: self.max_bid_lock_duration,
+            min_starting_price: self.min_starting_price,
+            max_starting_price: self.max_starting_price,
+            min_creator_royalty: self.min_creator_royalty,
+            max_creator_royalty: self.max_creator_royalty,
+            min_bid_step: self.min_bid_step,
+            max_bid_step: self.max_bid_step,
+            platform_fee_percentage: self.platform_fee_percentage,
+            auction_starts_at_max_delta: self.auction_starts_at_max_delta,
+            max_auction_whitelist_length: self.max_auction_whitelist_length,
+        }
```

Comment:
Probably worth refactoring this into a separate method

Reply:
see 2228e1ed689475a41645a46f0da104a98894cdaf it's similar to the project token pallet now